apps/photos/src/services/collectionService.ts

```diff
@@ -1260,3 +1260,31 @@ export async function unhideToCollection(
         throw e;
     }
 }
+
+export const userIdtoEmail = async (): Promise<Map<number, string>> => {
+    try {
+        const collection = await getLocalCollections();
+        const user: User = getData(LS_KEYS.USER);
+        const emailMapping = new Map<number, string>();
+        collection.map((item) => {
+            const { owner } = item;
+            const { sharees } = item;
+
+            if (user.id !== owner.id && owner.email) {
+                emailMapping.set(owner.id, owner.email);
+            }
+
+            if (sharees) {
+                sharees.map((item) => {
+                    if (item.id !== user.id)
+                        emailMapping.set(item.id, item.email);
+                });
+            }
+        });
+        const userIdToEmail = emailMapping;
+        return userIdToEmail;
+    } catch (e) {
+        logError('Error Mapping UserId to email:', e);
+        throw e;
```

Comment:
To avoid breaking UI, let's return emptyMap.
Or we can define const emailMapping = new Map<number, string>(); before try catch and just return that.

Add a comment saying to avoid breaking UI, we are swallowing the error after logging.